tests/default.nix

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -I nixpkgs=. ./. --pure --run true
```

Comment:
Not a fan of this. How about a simple `nix-build` that builds a script?

Reply:
I have rolled back to before the point where the top-level entry point was added.  I've also linked this comment here:

- https://github.com/NixOS/nixpkgs/issues/272591

so we don't lose this feedback.